tests/ptvsd/server/test_flask.py

```diff
@@ -29,56 +29,54 @@ class lines:
     app_py = code.get_marked_line_numbers(paths.app_py)
 
 
-def _initialize_session(session, multiprocess=None, exit_code=0):
-    if multiprocess:
-        pytest.skip("https://github.com/microsoft/ptvsd/issues/1706")
-
-    env = {
-        "FLASK_APP": paths.app_py,
-        "FLASK_ENV": "development",
-        "FLASK_DEBUG": "1" if multiprocess else "0",
-    }
-    if platform.system() != "Windows":
-        locale = "en_US.utf8" if platform.system() == "Linux" else "en_US.UTF-8"
-        env.update({"LC_ALL": locale, "LANG": locale})
-
-    args = ["run"]
-    if not multiprocess:
-        args += ["--no-debugger", "--no-reload", "--with-threads"]
-    args += ["--port", str(flask.port)]
-
-    session.expected_exit_code = exit_code
-    session.configure(
-        "module",
-        "flask",
-        cwd=paths.flask1,
-        jinja=True,
-        subProcess=multiprocess,
-        args=args,
-        env=env,
-    )
-
-
-@pytest.mark.parametrize(
-    "start_method", [runners.launch, runners.attach_by_socket["cli"]]
-)
+@pytest.fixture
+@pytest.mark.parametrize("run", [runners.launch, runners.attach_by_socket["cli"]])
+def start_flask(run):
+    def start(session, multiprocess=False):
+        if multiprocess:
+            pytest.skip("https://github.com/microsoft/ptvsd/issues/1706")
+
+        env = {
+            "FLASK_APP": paths.app_py,
+            "FLASK_ENV": "development",
+            "FLASK_DEBUG": "1" if multiprocess else "0",
+        }
+        if platform.system() != "Windows":
+            locale = "en_US.utf8" if platform.system() == "Linux" else "en_US.UTF-8"
+            env.update({"LC_ALL": locale, "LANG": locale})
+
+        # No clean way to kill Flask server, expect non-zero exit code
+        session.expected_exit_code = some.int
```

Comment:
Let's move this bit down after `args` and before `run`, so that it's not in the middle of editing `session.config`.